src/rgw/rgw_lc_s3.cc

```diff
@@ -128,15 +128,20 @@ void LCFilter_S3::dump_xml(Formatter *f) const
 
 void LCFilter_S3::decode_xml(XMLObj *obj)
 {
-  XMLObj *o = obj->find_first("And");
-  bool single_cond = false;
+  /*
+   * The prior logic here "looked for" and And element, but did not
+   * structurally parse the Filter clause (and incorrectly rejected
+   * the base case where a Prefix and one Tag were supplied).  It
+   * could not reject generally malformed Filter syntax.
+   *
+   * Empty filters are allowed:
+   * https://docs.aws.amazon.com/AmazonS3/latest/dev/intro-lifecycle-rules.html
+   */
   int num_conditions = 0;
```

Comment:
I believe an older version of docs were forcing a single condition in the official docs at least, since it looks like that is no longer a necessity the whole single_cond logic can be dropped